### PR TITLE
Bundle firmware updater with software store

### PR DESCRIPTION
**Pull request by d-loose**
- Adds [firmware-updater](https://github.com/canonical/firmware-updater) as a separate app within the snap
- Runs firmware-notifier periodically (every three hours)

P.S. To test it locally change the timer to run every minute or so, e.g.
```diff
diff --git a/snap/snapcraft.yaml b/snap/snapcraft.yaml
index 9927adf..28a6401 100644
--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -131,7 +131,7 @@ apps:
     daemon: simple
     passthrough:
       daemon-scope: user
-    timer: '00:00-24:00/8'
+    timer: '00:00-24:00/1440'
     extensions: [gnome]
     plugs:
       - desktop
```
build the snap, and connect all plugs manually.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -50,13 +51,52 @@ parts:
...
-      sed -i -e 's|\(Icon=\).*|\1/bin/data/flutter_assets/assets/software.png|' \
+      sed -i -e 's|\(Icon=\).*|\1/software/bin/data/flutter_assets/assets/software.png|' \
           $CRAFT_PART_INSTALL/bin/data/flutter_assets/assets/software.desktop
+    organize:
+      bin: software/bin
+
+  firmware-updater-app:
+    after: [flutter-git]
+    plugin: nil
+    source: https://github.com/canonical/firmware-updater.git
```
Comment (kenvandine) [inline snap/snapcraft.yaml:62]: add source-tag

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -50,13 +51,52 @@ parts:
...
+      flutter build linux --release -v
+      mkdir -p $CRAFT_PART_INSTALL/bin/
+      cp -r build/linux/*/release/bundle/* $CRAFT_PART_INSTALL/bin/
+    organize:
+      bin: firmware-updater/bin
+
+  firmware-notifier:
+    after: [flutter-git]
+    plugin: nil
+    source: https://github.com/canonical/firmware-updater.git
```
Comment (kenvandine) [inline snap/snapcraft.yaml:79]: add source-tag

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -50,13 +51,52 @@ parts:
...
+      set -eux
+      dart pub get
+      mkdir -p $CRAFT_PART_INSTALL/bin/
+      dart compile exe bin/firmware_notifier.dart -o $CRAFT_PART_INSTALL/bin/firmware-notifier
+    organize:
+      bin: firmware-updater/bin
+
+  firmware-updater:
+    plugin: dump
+    source: https://github.com/canonical/firmware-updater.git
```
Comment (kenvandine) [inline snap/snapcraft.yaml:91]: add source-tag

**Review by kenvandine — CHANGES_REQUESTED** [review]
specify source-tag

**Comment by d-loose** [comment]
There's no versioning scheme for the firmware updater yet - is it time to introduce one?
I've specified the latest commit for now.

**Comment by d-loose** [comment]
Closing this for now. I'll rebase and reopen it later if we still want to ship the store and the firmware updater together.

**Pull request closed without merging**